### ISSUE TITLE
xmonad with a displayManager won't work since 20.03

### ISSUE THREAD

**Issue by PierreR** [rycee/home-manager#1180]
Since 20.03, it looks like it is a requirement to define `services.xserver.displayManager.defaultSession` to `none+xmonad` to get xmonad working with autoLogin. 

This setting *only* works if `xmonad` is already defined in configuration.nix using:

```
    services.xserver.windowManager.xmonad = true;
```

The issue is that I am defining the windowManager with home-manager:
```
 xsession = {
      enable = true;
      windowManager.xmonad.enable = true;
 };
```

I don't mind keeping a system configuration that uses `xterm` as the defaultSession but how can I change the value of `services.xserver.displayManager.defaultSession` to `none+xmonad` when the home-manager kicks in with the xmonad config ?

In other words, how can I make use of the displayManager with xmonad defined in the home-manager since 20.03 ?

Thanks for your help. I am totally confused by this.

**Comment by rycee** [comment]
I have
```
services.xserver.desktopManager.xterm.enable = true;
```
in my NixOS configuration with
```
xsession.windowManager.xmonad.enable = true;
```
in my HM configuration. This seems to work well in NixOS 20.03 as well. I haven't really looked into `services.xserver.displayManager.defaultSession` yet.

**Comment by PierreR** [comment]
Thanks.

The problem arises when you want to autoLogin using the displayManager. I am using `lightdm` (the default one).

Since 20.03 `autoLogin` seems to require `services.xserver.displayManager.defaultSession`

**Comment by rycee** [comment]
Right, I had a quick look at the related code in NixOS and don't quite understand how it all fits together. Would something like https://discourse.nixos.org/t/opening-i3-from-home-manager-automatically/4849/8 work?

**Comment by PierreR** [comment]
It looks like it works indeed !

Thanks !

**Comment by spacekitteh** [comment]
As an aside, but am I correct in assuming that if `home-manager` is installed on, say, Ubuntu, then it cannot manage display managers?
